### DISCUSSION TITLE
membrane_ffmpeg_plugin

### DISCUSSION THREAD

**Discussion by mat-hek** [membraneframework/membrane-core#711] [Ideas]
The idea is to build a plugin that would provide elements that would spawn FFmpeg via a port and pass the stream between it and the pipeline.

User story:

As a Membrane user, I’d like Membrane to be easily interoperable with FFmpeg, so that I can use Membrane with my existing FFmpeg script or use features that are available in FFmpeg and not in Membrane.

AC:

Package with a source, sink and filter (or a filter with dynamic, possibly absent inputs/outputs) that would provide FFmpeg interop

Elements should accept an FFmpeg command and maybe [ffmpex](https://github.com/talklittle/ffmpex) syntax too

We should manage the FFmpeg process and make sure it’s properly terminated. Maybe we can use ffmpex under the hood

**Comment by dmorn** [comment]
Hey @mat-hek! FYI We're using a combination of [Exile](https://github.com/akash-akya/exile/tree/v0.10.0) and FFmpeg in some filters and it rocks. Usually we combine those filters with a Membrane parser afterwards. I can provide some examples if you want.
